src/arena/cfr/action_generator.rs

```diff
@@ -146,18 +146,16 @@ impl ActionGenerator for BasicCFRActionGenerator {
         if all_in_ammount > game_state.current_round_bet() {
             // All-in, Bet all the money
             // Bet everything we have bet so far plus the remaining stack
-            res.push(AgentAction::Bet(all_in_ammount));
+            res.push(AgentAction::AllIn);
         }
         res
     }
 
-    fn action_to_idx(&self, game_state: &GameState, action: &AgentAction) -> usize {
-        if action == &AgentAction::Fold {
-            0
-        } else if action == &AgentAction::Bet(game_state.current_round_bet()) {
-            1
-        } else {
-            2
+    fn action_to_idx(&self, action: &AgentAction) -> usize {
+        match action {
+            AgentAction::Fold => 0,
+            AgentAction::Bet(_) => 1,
+            AgentAction::AllIn => 2,
```

Comment:
I assume that there was an issue where the all in was getting mis-classified. Is there some way that we can fix that without adding a new AgentAction ? Something like saying if the bet made the player have 0 left then it was an all in?

Reply:
Yeah, the ordering of do_bet and record_action messed it up. If a player went all in, do_bet would set the round's bet to that amount, but then action_to_idx would consider that a check/call because their bet was the current round's bet (and always would be when called in this order).

I think that the proper fix for this while keeping the actions limited to Fold and Bet would be to work on how do_bet and record_action are called. I can look into it.

Reply:
I don't think that we can rely on how much the player has left to get the index of the action.

Let's say each player has bet 10.0 so far. In a new round, player 1 goes all in, bringing the current round bet to 40.0. Player 0 _also_ bets 40, and now has nothing left. Did player 0 go all in? or did they call? With respect to the graph, "all in" means nothing. What really matters in terms of the game state is if a player folds, checks/calls, or raises (in the simplified poker that we have right here, going all in is the only way to raise, so they are one in the same). So what we really need to know when someone bets is "is their bet higher than the current bet?".

Reply:
> Did player 0 go all in? or did they call?

The historian cares about the intent of the agent. The state is there to try and put a value on the action that the agent is trying to do. So it's crucial that the action that the agent thinks its taking matches what the historian records.

Later there will even be actions that overlap. For example if some agent has an action that is bet 10% +/- 0.5% of the pot and they have an action that is bet minimium. It's possible that those overlap.

So I think we will have to pass the index around, but that's something to do in another PR. This one is solving the problem that's there.